test cases/common/193 dict/meson.build

```diff
@@ -33,3 +33,33 @@ d3 = d2
 d3 += {'e' : 'f'}
 assert(d3 == {'a' : 'b2', 'c' : 'd', 'e' : 'f'}, 'dict plusassign is not working')
 assert(d2 == {'a' : 'b2', 'c' : 'd'}, 'dict should be immutable')
+
+dict1 = {}
+
+# A variable to be used as a key
+testkey1 = 'myKey1'
+testkey2 = 'myKey2'
+
+# Add new entry using the variable
+dict1 += {testkey1 : 'myValue'}
+dict1 += {testkey2 : 42}
+
+# Test that the stored values are correct
+assert(dict1[testkey1] == 'myValue',
+       'Incorrect string value retrieved from dictionary - variable key')
+assert(dict1['myKey1'] == 'myValue',
+       'Incorrect string value retrieved from dictionary - literal key')
+assert(dict1[testkey2] == 42,
+       'Incorrect int value retrieved from dictionary - variable key')
+assert(dict1['myKey2'] == 42,
+       'Incorrect int value retrieved from dictionary - literal key')
+
+d = {testkey1 : 1}
+assert(d[testkey1] == 1,
+       'Incorrect string value retrieved from dictionary - variable key')
+assert(d['myKey1'] == 1,
+       'Incorrect string value retrieved from dictionary - literal key')
```

Comment:
1 is an int, not a string.

Now that the number of tests is growing (a Good Thing!), it's becoming more obvious how generic and vague this assert message is. My recommendation:
```
assert(dict1['myKey2'] == 42, 'dict1[\'myKey2\'] != 42')
```
As specific as possible, as short as possible and not error-prone at all.

Reply:
IMHO, assert() should not even need a message. They are most of the time useless, all you need is the line number where it failed. Ideally the interpreter should be smart enough to be able to take first arg code as string message, but I don't think that's possible currently.

Anyway, I fixed the message.

Reply:
> Ideally the interpreter should be smart enough to be able to take first arg code as string message,

Agreed 100%, in fact this is what I tried right before making that comment and why I recommend copying the first argument into the second.

> but I don't think that's possible currently.

It would require `assert` to be near the very top of the syntax tree.

Reply:
Yes, I gave it a try, but my conclusion is we need to special-case assert() in mparser.py, and that's kind of layer violation, so I'm not so sure. OTOH, in C assert is typically a preprocessor macro for that reason.

Reply:
Actually, found an interesting way of doing it: https://github.com/mesonbuild/meson/pull/6285